app/models/queue_item.rb

```diff
@@ -5,12 +5,30 @@ class QueueItem < ActiveRecord::Base
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
+  validates_numericality_of :position, {only_integer: true}
+
   def rating
-    review = Review.where(user_id: user.id, video_id: video.id).first
     review.rating if review
   end
 
+  def rating=(new_rating)
+    if review
+      review.update_column(:rating, new_rating)
+    else
+      review = Review.new(user: user, video: video, rating: new_rating)
+      review.save(validate: false)
+    end
+    
+  end
+
   def category_name
     category.name
   end
+
+  private
+
+  def review
+    @review ||= Review.where(user_id: user.id, video_id: video.id).first
```

Comment:
I know Kevin is using `.where(args).first` in solutions but that is a Rails 3 thing and from Rails 4 it is recommended to use `.find_by` as it is more efficient than `.where(args).first` when searching for a single record. If you search for collection use `.where`.